plutus-core/plutus-core/src/PlutusCore/Flat.hs

```diff
@@ -101,6 +102,10 @@ encodeConstant = safeEncodeBits constantWidth
 decodeConstant :: Get Word8
 decodeConstant = dBEBits8 constantWidth
 
+instance Flat Data where
+    encode = error "Implement me"
+    decode = error "Implement me"
```

Comment:
@michaelpj so what should go here? Care to implement that yourself? Or maybe let's let @kwxm do that? I don't really know how we use and implement `Flat` and `Serialize` instances.

Reply:
So `Data` is "really" a subset of CBOR. We could try and give it a proper `Flat` instance, or we could just exploit the fact that it has a well-defined CBOR encoding already, and just say something like

```
instance Flat Data where
    encode = encode @ByteString . CBOR.serialise
    decode = CBOR.deserialise . decode @ByteString
```

That seems fine to me, tbh. For bonus points, define a `AsCBOR` newtype and write `instance CBOR.Serialise a => Flat (AsCBOR a)` for `DerivingVia`.